### PR TITLE
Feature object pooling generic

### PR DESCRIPTION
**Pull request by vijit101**
- Added Object pooler generic code
- Added object pooler normal code

### CODE REVIEW

File: Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+    private List<PooledItem<T>> PooledItems;
+
+    public virtual T GetPoolItem()
+    {
+        if(PooledItems.Count > 0)
+        {
+            PooledItem<T> poolobj = PooledItems.Find(i => i.m_isUsed == false);// foreach(item i in pooleditem){if(!item.isused)return item.item}
+            poolobj.m_isUsed = true;
+            if(poolobj.m_poolItem != null)
```
Comment (mayankgrover) [inline Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs]: m_ is not a C# convention please remove these from everywhere

File: Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        PooledItem<T> returnItem = PooledItems.Find(i => i.m_poolItem == ReturnedItem);//If not working use .equals
+        returnItem.m_isUsed = false;
+        return;
+    }
+
+
+
+
+#pragma warning disable CS0693 // Type parameter has the same name as the type parameter from outer type
```
Comment (mayankgrover) [inline Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs]: why do you need this?

File: Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ObjectPoolGeneric<T> : Singletongeneric<ObjectPoolGeneric<T>> where T:class
```
Comment (mayankgrover) [inline Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs]: namespace

File: Assets/Scripts/Utils & Misc/ObjectPooler.cs [hunk 4]
```diff
@@ -1,76 +1,76 @@
...
-    public int amountToPool;
-    public GameObject objectToPool;
-    public bool shouldExpand;
-}
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+public class ObjectPoolGeneric : MonoBehaviour
```
Comment (mayankgrover) [inline Assets/Scripts/Utils & Misc/ObjectPooler.cs:7]: what is this?

File: Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs (hunk 1 above)
Comment (vijit101) [reply Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs]: sure will

File: Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs (hunk 2 above)
Comment (vijit101) [reply Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs]: visual studio showed me potential fixed so just clicked it

File: Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs (hunk 3 above)
Comment (vijit101) [reply Assets/Scripts/Utils & Misc/ObjectPoolGeneric.cs]: done

**Pull request closed without merging**